File: embeds/memberModeration.ts
import { GuildMember, EmbedBuilder } from "discord.js";
import { ACTIONS, ATTR_PRETTY, Reason } from "../clients/constants.js";

export default function embedMemberModeration(
  member: GuildMember,
  action: number,
  reasons: Reason[],
) {
  const embed = new EmbedBuilder()
    .setTitle(`Flagged Psycho-Pass of ${member.user.username}`)
    .setAuthor({
      name: "Sibyl System",
      iconURL: member.client.user.avatarURL()!,
    })
    .setTimestamp()
    .setFooter({ text: member.id });

  if (ACTIONS[action] === "BAN")
    embed.setDescription(`Banned ${member.user.username}`);
  else if (ACTIONS[action] === "KICK")
    embed.setDescription(`Kicked ${member.user.username}`);
  else if (ACTIONS[action] === "MUTE")
    embed.setDescription(`Muted ${member.user.username}`);
  else if (action >= ACTIONS.indexOf("NOTIFY"))
    embed.setDescription("Notified Moderators");

  for (const reason of reasons) {
    embed.addFields(
      { name: "\u200B", value: "\u200B" },
      {
        name: ATTR_PRETTY[reason.attribute as keyof typeof ATTR_PRETTY],
        value: reason.score.toString(),
        inline: true,
      },
      {
        name: "Threshold",
        value: reason.threshold.toString(),
        inline: true,
      },
    );
  }

  return embed;
}

File: embeds/member_moderation.py
from datetime import datetime, UTC
from discord import Embed, EmbedAuthor, EmbedFooter, Member
from clients.constants import ACTION, ATTR_PRETTY


def embed_member_moderation(
    member: Member, action: ACTION, reasons: list[tuple]
) -> Embed:
    embed = Embed(
        title=f"flagged psycho-pass of {member.name}",
        author=EmbedAuthor(name="sibylmod", icon_url=member.avatar.url),
        timestamp=datetime.now(UTC),
        footer=EmbedFooter(text=member.id),
    )

    if action is ACTION.BAN:
        embed.description = f"banned {member.mention}"
    elif action is ACTION.KICK:
        embed.description = f"kicked {member.mention}"
    elif action is ACTION.MUTE:
        embed.description = f"muted {member.mention}"
    elif action >= ACTION.NOTIFY:
        embed.description = "notified moderators"

    for reason in reasons:
        attribute, score, threshold = reason
        embed.add_field(name="\u200B", value="\u200B")
        embed.add_field(name=ATTR_PRETTY[attribute], value=score, inline=True)
        embed.add_field(name="threshold", value=threshold, inline=True)

    return embed
